File: app/src/main/java/com/vondear/tools/activity/ActivityCreateQRCode.java
package com.vondear.tools.activity;

import android.annotation.SuppressLint;
import android.os.Bundle;
import android.os.Handler;
import android.os.Message;
import android.view.View;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.vondear.rxtools.RxBarTool;
import com.vondear.rxtools.activity.ActivityBase;
import com.vondear.rxtools.view.RxBarCode;
import com.vondear.rxtools.view.RxQRCode;
import com.vondear.rxtools.view.RxTitle;
import com.vondear.tools.R;

import butterknife.BindView;
import butterknife.ButterKnife;

import static com.vondear.tools.R.id.iv_linecode;

/**
 * @author vondear
 */
public class ActivityCreateQRCode extends ActivityBase implements View.OnClickListener {

    private static android.os.Handler Handler = new Handler();
    private static Runnable mRunnable = null;
    @BindView(R.id.rx_title)
    RxTitle mRxTitle;
    @BindView(iv_linecode)
    ImageView mIvLinecode;
    @BindView(R.id.iv_code)
    ImageView mIvCode;
    @BindView(R.id.imageView1)
    ImageView mImageView1;
    @BindView(R.id.textView2)
    TextView mTextView2;
    @BindView(R.id.tv_time_second)
    TextView mTvTimeSecond;
    @BindView(R.id.ll_refresh)
    LinearLayout mLlRefresh;
    @SuppressLint("HandlerLeak")
    Handler mHandler = new Handler() {
        public void handleMessage(Message msg) {
            super.handleMessage(msg);
            switch (msg.what) {
                case 60000:
                    initData();
                    break;
                default:
                    break;
            }
        }
    };
    private int time_second = 0;
    private int second = 60;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        RxBarTool.noTitle(mContext);
        setContentView(R.layout.activity_create_qrcode);
        RxBarTool.setTransparentStatusBar(mContext);
        ButterKnife.bind(this);
        initView();
        initData();
        AuthCode(mTvTimeSecond, second);
    }

    private void AuthCode(final TextView view, final int timeSecond) {

        mRunnable = new Runnable() {
            int mSumNum = timeSecond;

            @Override
            public void run() {
                Handler.postDelayed(mRunnable, 1000);
                view.setText(mSumNum + "");
                view.setEnabled(false);
                mSumNum--;
                if (mSumNum < 0) {
                    view.setText(0 + "");
                    view.setEnabled(true);
                    Message message = new Message();
                    message.what = 60000;
                    mHandler.sendMessage(message);
                    // 干掉这个定时器，下次减不会累加
                    Handler.removeCallbacks(mRunnable);
                    AuthCode(mTvTimeSecond, second);
                }
            }

        };
        Handler.postDelayed(mRunnable, 1000);
    }

    private void initView() {
        mRxTitle.setLeftFinish(mContext);
        mRxTitle.setTitle("动态生成码");
        mLlRefresh.setOnClickListener(this);
    }

    private void initData() {
        // TODO Auto-generated method stub
        RxQRCode.createQRCode("时间戳:" + System.currentTimeMillis(), 800, 800, mIvCode);
        mIvLinecode.setImageBitmap(RxBarCode.createBarCode("" + System.currentTimeMillis(), 1000, 300));
    }

    @Override
    public void onClick(View arg0) {
        switch (arg0.getId()) {
            case R.id.ll_refresh:
                Handler.removeCallbacks(mRunnable);
                initData();
                mTvTimeSecond.setText(second + "");
                AuthCode(mTvTimeSecond, second);
                break;
            default:
                break;
        }
    }
}

File: RxDemo/src/main/java/com/tamsiree/rxdemo/activity/ActivityCreateQRCode.kt
package com.tamsiree.rxdemo.activity

import android.annotation.SuppressLint
import android.os.Bundle
import android.os.Handler
import android.os.Message
import android.view.View
import android.widget.TextView
import com.tamsiree.rxdemo.R
import com.tamsiree.rxfeature.tool.RxBarCode
import com.tamsiree.rxfeature.tool.RxQRCode
import com.tamsiree.rxkit.RxBarTool.noTitle
import com.tamsiree.rxkit.RxBarTool.setTransparentStatusBar
import com.tamsiree.rxkit.RxDeviceTool.setPortrait
import com.tamsiree.rxui.activity.ActivityBase
import kotlinx.android.synthetic.main.activity_create_qrcode.*

/**
 * @author tamsiree
 */
class ActivityCreateQRCode : ActivityBase(), View.OnClickListener {

    @SuppressLint("HandlerLeak")
    var mHandler: Handler = object : Handler() {
        override fun handleMessage(msg: Message) {
            super.handleMessage(msg)
            when (msg.what) {
                60000 -> initData()
            }
        }
    }
    private val time_second = 0
    private val second = 60
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        noTitle(mContext)
        setContentView(R.layout.activity_create_qrcode)
        setTransparentStatusBar(mContext)
        setPortrait(this)
    }

    private fun AuthCode(view: TextView?, timeSecond: Int) {
        mRunnable = object : Runnable {
            var mSumNum = timeSecond
            override fun run() {
                Handler.postDelayed(mRunnable, 1000)
                view!!.text = mSumNum.toString() + ""
                view.isEnabled = false
                mSumNum--
                if (mSumNum < 0) {
                    view.text = 0.toString() + ""
                    view.isEnabled = true
                    val message = Message()
                    message.what = 60000
                    mHandler.sendMessage(message)
                    // 干掉这个定时器，下次减不会累加
                    Handler.removeCallbacks(mRunnable)
                    AuthCode(tv_time_second, second)
                }
            }
        }
        Handler.postDelayed(mRunnable, 1000)
    }

    override fun initView() {
        rx_title.setLeftFinish(mContext)
        rx_title.title = "动态生成码"
        ll_refresh.setOnClickListener(this)
    }

    override fun initData() {
        RxQRCode.createQRCode("时间戳:" + System.currentTimeMillis(), 800, 800, iv_code)
        iv_linecode.setImageBitmap(RxBarCode.createBarCode("" + System.currentTimeMillis(), 1000, 300))
        AuthCode(tv_time_second, second)
    }

    override fun onClick(arg0: View) {
        when (arg0.id) {
            R.id.ll_refresh -> {
                Handler.removeCallbacks(mRunnable)
                initData()
                tv_time_second.text = second.toString() + ""
                AuthCode(tv_time_second, second)
            }
            else -> {
            }
        }
    }

    companion object {
        private val Handler = Handler()
        private var mRunnable: Runnable? = null
    }
}